### PR TITLE
Added return types for Model class

### PR DESCRIPTION
**Pull request by christinacai123**

### CODE REVIEW

**Review by christinacai123 — COMMENTED** [review]
fixed circular import error

File: openmc/model/model.py [hunk 1]
```diff
@@ -132,14 +134,14 @@ def _materials_by_id(self):
...
-    def _cells_by_id(self):
+    def _cells_by_id(self) -> dict:
         """Dictionary mapping cell ID --> cell"""
         cells = self.geometry.get_all_cells()
         return {cell.id: cell for cell in cells.values()}
 
     @property
     @lru_cache(maxsize=None)
-    def _cells_by_name(self):
+    def _cells_by_name(self) -> dict:
```
Comment (paulromano) [inline openmc/model/model.py:144]: You can use typing.Dict to make this more specific
```suggestion
    def _cells_by_name(self) -> Dict[int, openmc.Cell]:
```

File: openmc/model/model.py [hunk 2]
```diff
@@ -7,6 +8,7 @@
 from tempfile import NamedTemporaryFile
 import warnings
 from xml.etree import ElementTree as ET
+from typing import Optional
```
Comment (paulromano) [inline openmc/model/model.py:11]:
```suggestion
from typing import Optional, Dict
```

File: openmc/model/model.py [hunk 3]
```diff
@@ -152,7 +154,7 @@ def _cells_by_name(self):
 
     @property
     @lru_cache(maxsize=None)
-    def _materials_by_name(self):
+    def _materials_by_name(self) -> dict:
```
Comment (paulromano) [inline openmc/model/model.py:157]:
```suggestion
    def _materials_by_name(self) -> Dict[int, openmc.Material]:
```

**Review by paulromano — CHANGES_REQUESTED** [review]

File: openmc/model/model.py [hunk 4]
```diff
@@ -604,7 +606,7 @@ def run(self, particles=None, threads=None, geometry_debug=False,
         this method creates the XML files and runs OpenMC via a system call. In
         both cases this method returns the path to the last statepoint file
         generated.
-
+
```
Comment (paulromano) [inline openmc/model/model.py:609]:
```suggestion

```

**Review by paulromano — CHANGES_REQUESTED** [review]

**Review by paulromano — APPROVED** [review]

**Pull request merged**